### PR TITLE
Delete userbyid

### PR DESCRIPTION
**Pull request by tariqjavid**
# Description

   i have added Delete user by  id component
Fixes # (issue)

# How to test? postman

# Checklist

- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/routes/api-router.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const modulesRouter = require('./modules.router');
 
 // messages router imports
 const messagesRouter = require('./messages-router');
+const userRouter = require('./userDeleteById-router');
```
Comment (benna100) [inline src/server/api/routes/api-router.js:10]: The user router has already been added, merge develop into your branch. The delete user should lay under the user router.

**Review by benna100 — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/users-controller.js [hunk 2]
```diff
@@ -2,12 +2,24 @@ const knex = require('../../config/db');
 
 const getUserById = async (id) => {
   try {
-    const user = await knex('users').select('user_name', 'id').where({ id });
+    const user = await knex('users')
+      .select('user_name', 'id')
+      .where({ id });
```
Comment (benna100) [inline src/server/api/controllers/users-controller.js:7]: Again Tariq this is not related to your PR!

File: src/server/api/routes/users-router.js [hunk 3]
```diff
@@ -48,6 +48,14 @@ router.get('/', (req, res, next) => {
...
 });
 
+// user delete by id
+router.delete('/:id', (req, res, next) => {
+  usersController
+    .userDeleteById(req.params.id)
+    .then((result) => res.json(result))
+    .catch(next);
+});
+
```
Comment (benna100) [inline src/server/api/routes/users-router.js:58]: Remember the swagger documentation

**Review by benna100 — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/users-controller.js (hunk 2 above)
Comment (tariqjavid) [reply src/server/api/controllers/users-controller.js:7]: ,i just took changes from develop  then merge in feature branch,  i don't know  how is come!

**Comment by benna100** [comment]
Whats going on here tariq. Did you not just need to add the changes you had already made?

This PR has been open for 7 days now, lets get it merged soon

**Comment by tariqjavid** [comment]
i was waiting for you to merge the pr.I had done changes ,please let me know that if need more changes
